kernel/src/fs/procfs/pid/task/environ.rs

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{
+    fs::{
+        procfs::template::{FileOps, ProcFileBuilder},
+        utils::Inode,
+    },
+    prelude::*,
+    Process,
+};
+
+/// Represents the inode at `/proc/[pid]/task/[tid]/environ` (and also `/proc/[pid]/environ`).
+pub struct EnvironFileOps(Arc<Process>);
+
+impl EnvironFileOps {
+    pub fn new_inode(process_ref: Arc<Process>, parent: Weak<dyn Inode>) -> Arc<dyn Inode> {
+        ProcFileBuilder::new(Self(process_ref))
+            .parent(parent)
+            .build()
+            .unwrap()
+    }
+}
+
+impl FileOps for EnvironFileOps {
+    fn data(&self) -> Result<Vec<u8>> {
+        let envp = self.0.init_stack_reader().envp()?;
+        let mut environ_output = Vec::new();
+        for var in envp {
+            environ_output.extend_from_slice(var.as_bytes());
+            environ_output.push(b'\0');
```

Comment:
`var` is `CString` with ending nul byte, so these two lines can be simplified as
```rust
environ_output.extend_from_slice(var.as_bytes_with_nul());
```